tests/app/dao/test_returned_letters_dao.py

```diff
@@ -0,0 +1,114 @@
+from datetime import datetime, timedelta
+
+from freezegun import freeze_time
+
+from app.dao.returned_letters_dao import insert_or_update_returned_letters, get_returned_letter_summary
+from app.models import ReturnedLetter
+from tests.app.db import create_notification, create_notification_history, create_returned_letter
+
+
+def test_insert_or_update_returned_letters_inserts(sample_letter_template):
+    notification = create_notification(template=sample_letter_template,
+                                       reference='ref1')
+    history = create_notification_history(template=sample_letter_template,
+                                          reference='ref2')
+
+    assert ReturnedLetter.query.count() == 0
+
+    insert_or_update_returned_letters(['ref1', 'ref2'])
+
+    returned_letters = ReturnedLetter.query.all()
+
+    assert len(returned_letters) == 2
+    returned_letters_ = [x.notification_id for x in returned_letters]
+    assert notification.id in returned_letters_
+    assert history.id in returned_letters_
+
+
+def test_insert_or_update_returned_letters_updates(sample_letter_template):
+    notification = create_notification(template=sample_letter_template,
+                                       reference='ref1')
+    history = create_notification_history(template=sample_letter_template,
+                                          reference='ref2')
+
+    assert ReturnedLetter.query.count() == 0
+    with freeze_time('2019-12-09 13:30'):
+        insert_or_update_returned_letters(['ref1', 'ref2'])
+        returned_letters = ReturnedLetter.query.all()
+        assert len(returned_letters) == 2
+        for x in returned_letters:
+            assert x.reported_at == datetime(2019, 12, 9, 13, 30)
+            assert x.notification_id in [notification.id, history.id]
+
+    with freeze_time('2019-12-10 14:20'):
+        insert_or_update_returned_letters(['ref1', 'ref2'])
+
+        returned_letters = ReturnedLetter.query.all()
+        assert len(returned_letters) == 2
+        for x in returned_letters:
+            assert x.reported_at == datetime(2019, 12, 10, 14, 20)
+            assert x.notification_id in [notification.id, history.id]
+
+
+def test_insert_or_update_returned_letters_when_no_notification(sample_letter_template):
```

Comment:
This test could use the `notify_db_session` fixture instead of `sample_letter_template`, which isn't used